speculation-rules.bs

```diff
@@ -0,0 +1,275 @@
+
+Since the parsing behavior of the `<script>` element has long been part of HTML, any modern HTML parser would not construct any non-text children of the element. There is thus a low risk of other text hidden inside a `<script>` element with `type="speculationrules"` which is parsed as part of the script content by compliant HTML implementations but as HTML tags by others.
+
+Authors should, however, still escape any potentially attacker-controlled content inserted into speculation rules. In particular, it may be necessary to escape JSON syntax as well as, if the speculation rules are in an inline `<script>` tag, the closing `</script>` tag. [[CSP]] is a useful additional mitigation for vulnerabilities of this type.
+
+<div class="issue">Expand this section once externally loaded (via "`src`") speculation rules are specified.</div>
+
+<h3 id="security-ip-anonymization">IP anonymization</h3>
+
+This specification allows authors to request prefetch traffic using IP anonymization technology provided by the user agent. The details of this technology are not a part of this specification; nonetheless some general principles apply.
+
+To the extent IP anonymization is implemented using a proxy service, it is advisable to minimize the information available to the service operator and other entities on the network path. This likely involves, at a minimum, the use of [[TLS]] for the connection.
+
+Site operators should be aware that, similar to virtual private network (VPN) technology, the client IP address seen by the HTTP server may not exactly correspond to the user's actual network provider or location, and a traffic for multiple distinct subscribers may originate from a single client IP address. This may affect site operators' security and abuse prevention measures. IP anonymization measures may make an effort to use an egress IP address which has a similar geolocation or is located in the same jurisdiction as the user, but any such behavior is particular to the user agent and not guaranteed by this specification.
+
+<h2 id="privacy-considerations">Privacy considerations</h2>
+
+<h3 id="privacy-heuristics">Heuristics</h3>
+
+Because the candidate prefetches and other actions are not required, the user agent can use heuristics to determine which actions would be best to execute. Because it may be observable to the document whether actions were executed, user agents should take care to protect privacy when making such decisions -- for instance by only using information which is already available to the origin. To the extent these heuristics depend on any persistent state, that state should be erased whenever other site data is erased, possibly including time-based expiry.
```

Comment:
"possibly including time-based expiry" doesn't make too much sense to me. In particular it doesn't seem to be "included" in erasing site data.

Maybe "Additionally, the user agent might want to clear such state after some time" or similar?

Reply:
Hmm, how about:

```suggestion
Because the candidate prefetches and other actions are not required, the user agent can use heuristics to determine which actions would be best to execute. Because it may be observable to the document whether actions were executed, user agents should take care to protect privacy when making such decisions -- for instance by only using information which is already available to the origin. If these heuristics depend on any persistent state, that state should be erased whenever the user erases site data or the user agent clears it after some time.
```

Reply:
"that state should be erased whenever ... the user agent clears it after some time" seems tautological.

Reply:
Split it into two sentences to try and make it clearer, hopefully. Coarsely this is trying to say:
- other site data is cleared by user => this site data must be cleared
- other site data is cleared by expiry => this site data must be cleared

Reply:
Looks good!